### ISSUE TITLE
[Integration][JS] JS implementation is slow to build

### ISSUE THREAD

**Issue by pitrou** [apache/arrow#39118]
### Describe the enhancement requested

The JavaScript build in the integration tests is a bit slow. Here on a 12-core CPU:
```console
# time /arrow/ci/scripts/js_build.sh /arrow /build
[...]

real	1m8.027s
user	8m26.377s
sys	0m55.688s
```

I don't understand much of what it is doing, but it _seems_ it may be compiling the TypeScript sources with several different compiler options and/or versions, and also linting the sources. Given that this is the Integration build, the build could probably be slimmed down.

### Component(s)

Continuous Integration, Integration, JavaScript

**Comment by pitrou** [comment]
cc @domoritz @trxcllnt

**Comment by domoritz** [comment]
Sorry, by js versions I meant (imprecisely) module types like esm and cjs, and ecma script versions like es5, es2015, and esNext.

**Comment by pitrou** [comment]
Yes... but again, the integration tests by construction run only with a single version, even though several may be compiled. Just like they run with a single JVM version, a single C++ compiler, etc.
